src/content/profile-data.js

```diff
@@ -131,9 +131,24 @@ export function defaultThreadOrder(threads: Thread[]) {
   return threadOrder;
 }
 
-export function filterThreadToJSOnly(thread: Thread) {
-  return timeCode('filterThreadToJSOnly', () => {
-    const { stackTable, funcTable, frameTable, samples } = thread;
+export function filterThreadByImplementation(thread: Thread, implementation: string): Thread {
+  const { funcTable } = thread;
+
+  switch (implementation) {
+    case 'cpp':
+      return _filterThreadByFunc(thread, funcIndex => !funcTable.isJS[funcIndex]);
+    case 'js':
+      return _filterThreadByFunc(thread, funcIndex => funcTable.isJS[funcIndex]);
+  }
```

Comment:
idea: have a map (an object-as-a-map) to define the filters:
```javascript

const implementationFilters = {
  cpp: (funcTable, funcIndex) => !funcTable.isJS[funcIndex],
  js: (funcTable, funcIndex) => funcTable.isJS[funcIndex],
};

function filterThreadByImplementation(...) {
  ...
  return _filterThreadByFunc(thread, implementationFilters[implementation].bind(null, funcTable));
}
```
Another possibility is _filterThreadByFunc could pass the funcTable instead of binding.

As said Markus, in the future we'll likely want to chain several filters, this would also makes it easier.